language-support/java/codegen/src/main/scala/com/digitalasset/daml/lf/codegen/CodeGenRunner.scala

```diff
@@ -40,76 +125,71 @@ object CodeGenRunner extends StrictLogging {
+    val executionContext: ExecutionContextExecutorService = createExecutionContext()
+    val result = codegen.runWith(executionContext)
+    Await.result(result, 10.minutes)
+    executionContext.shutdownNow()
 
-    val (interfaces, pkgPrefixes) = collectDamlLfInterfaces(conf)
-    generateCode(interfaces, conf, pkgPrefixes)(ec)
-    val _ = executor.shutdownNow()
+    ()
   }
 
-  private[codegen] def collectDamlLfInterfaces(
-      conf: Conf
-  ): (Seq[Interface], Map[PackageId, String]) = {
-    val interfacesAndPrefixes = conf.darFiles.toList.flatMap { case (path, pkgPrefix) =>
-      val file = path.toFile
-      // Explicitly calling `get` to bubble up any exception when reading the dar
-      val dar = DarParser.assertReadArchiveFromFile(file)
-      dar.all.map { archive =>
-        val (errors, interface) = InterfaceReader.readInterface(archive)
-        if (!errors.equals(Errors.zeroErrors)) {
-          throw new RuntimeException(
-            InterfaceReader.InterfaceReaderError.treeReport(errors).toString
-          )
-        }
-        logger.trace(s"Daml-LF Archive decoded, packageId '${interface.packageId}'")
-        (interface, interface.packageId -> pkgPrefix)
+  private def createExecutionContext(): ExecutionContextExecutorService =
+    ExecutionContext.fromExecutorService(
+      Executors.newFixedThreadPool(
+        Runtime.getRuntime.availableProcessors(),
+        new ThreadFactory {
+          val n = new AtomicInteger(0)
+          override def newThread(r: Runnable): Thread = {
+            val t = new Thread(r)
+            t.setDaemon(true)
+            t.setName(s"java-codegen-${n.getAndIncrement}")
+            t
+          }
+        },
+      )
+    )
+
+  private[codegen] def decodeDarAt(path: Path): Seq[Interface] =
+    for (archive <- DarParser.assertReadArchiveFromFile(path.toFile).all) yield {
+      val (errors, interface) = Interface.read(archive)
+      if (!errors.equals(Errors.zeroErrors)) {
+        val message = InterfaceReader.InterfaceReaderError.treeReport(errors).toString
+        throw new RuntimeException(message)
       }
+      logger.trace(s"Daml-LF Archive decoded, packageId '${interface.packageId}'")
+      interface
     }
 
-    val interfaces = interfacesAndPrefixes.map(_._1)
-    val environmentInterface =
-      EnvironmentInterface.fromReaderInterfaces(interfaces.head, interfaces.tail: _*)
-    val prefixes = interfacesAndPrefixes.collect { case (_, (key, Some(value))) =>
-      (key, value)
-    }.toMap
-    val fullyResolvedInterfaces =
-      interfaces.map(
-        _.resolveChoicesAndFailOnUnresolvableChoices(environmentInterface.astInterfaces)
-      )
-    (fullyResolvedInterfaces, prefixes)
-  }
+  private[codegen] def resolveChoices(
+      environmentInterface: EnvironmentInterface
+  ): Interface => Interface =
+    _.resolveChoicesAndFailOnUnresolvableChoices(environmentInterface.astInterfaces)
 
-  private[CodeGenRunner] def generateFile(
```

Comment:
This method was unused.

Reply:
The diff interleaves the removal of this method with other methods I added, hope its not too unreadable.